### ISSUE TITLE
Error: Unexpected token (Note that you need plugins to import files that are not JavaScript)

### ISSUE THREAD

**Issue by hotamateurxxx** [rollup/plugins#1083]
- Rollup Plugin Name: @rollup/plugin-typescript
- Rollup Plugin Version: ^8.3.0
- Rollup Version: ^2.63.0
- TypeScript Version: ^4.5.4
- Node Version: v17.2.0
- Operating System (or Browser): Windows 10
- Link to reproduction: https://github.com/hotamateurxxx/rollup-typescript-bug

### Expected Behavior
Successfull build.

### Actual Behavior
```CMD
yarn build

...

Error: Unexpected token (Note that you need plugins to import files that are not JavaScript)
    at error (D:\Projects\github.com\rollup-typescript-bug\node_modules\rollup\dist\shared\rollup.js:158:30)
    at Module.error (D:\Projects\github.com\rollup-typescript-bug\node_modules\rollup\dist\shared\rollup.js:12423:16)
    at Module.tryParse (D:\Projects\github.com\rollup-typescript-bug\node_modules\rollup\dist\shared\rollup.js:12826:25)
    at Module.setSource (D:\Projects\github.com\rollup-typescript-bug\node_modules\rollup\dist\shared\rollup.js:12729:24)
    at ModuleLoader.addModuleSource (D:\Projects\github.com\rollup-typescript-bug\node_modules\rollup\dist\shared\rollup.js:22202:20)
```

### Additional Information

Got closed issue [ Typescript plugin 4.0.0 Error: Unexpected token (Note that you need plugins to import files that are not JavaScript) #287](https://github.com/rollup/plugins/issues/287) with plugin version that should to contain fix:

`node_modules\@rollup\plugin-typescript\CHANGELOG.md`
> ## v8.2.1
> 
> _2021-03-26_
> 
> ### Bugfixes
> 
> - fix: bump TypeScript version (#818)
> - fix: update readme and peerDeps version (#830)

**Comment by shellscape** [comment]
Please update your issue with the full template information.

**Comment by hotamateurxxx** [comment]
Update reproduction repo. Bug cause in case when you need to push build scripts into subfolder.

@shellscape, is that issue corresponds to full template information now?

**Comment by xenobytezero** [comment]
Rollup Plugin Name: @rollup/plugin-typescript
Rollup Plugin Version: ^8.3.1
Rollup Version: ^2.68.0
TypeScript Version: ^4.5.5
Node Version: v16.13.2
Operating System (or Browser): Windows 10

I can confirm that I am also getting this issue for the same reasons as the closed issue #287 , where the filter is not picking up the file in the `load()` hook.
